go/ethadapter/geth_rpc_client.go

```diff
@@ -25,8 +26,9 @@ import (
 )
 
 const (
-	connRetryMaxWait  = 10 * time.Minute // after this duration, we will stop retrying to connect and return the failure
-	connRetryInterval = 500 * time.Millisecond
+	connRetryMaxWait        = 10 * time.Minute // after this duration, we will stop retrying to connect and return the failure
```

Comment:
shouldn't this be in block time units as well?

Reply:
This is more to do with establishing a connection with the L1 RPC endpoint so not sure blocktime is as important. I don't think we handle this case well though, 10mins feels a bit arbitrary. Feels very relevant to that pool of L1 URLs we talked about earlier, I'll add a comment on that PR to rethink how we handle and configure this timeout.